drivers/lis2dh12/lis2dh12.c

```diff
@@ -187,6 +194,72 @@ int lis2dh12_read(const lis2dh12_t *dev, int16_t *data)
     return LIS2DH12_OK;
 }
 
+int lis2dh12_set_int(const lis2dh12_t *dev, const lis2dh12_int_params_t *params, uint8_t int_line)
+{
+    gpio_t pin = dev->p->int_pin[int_line - 1];
```

Comment:
You want to add an `assert((int_line > 0) && (int_line  < 3));` here.
Otherwise `gpio_init_int()` can access invalid memory.